cli/azd/pkg/project/scaffold_gen.go

```diff
@@ -445,173 +453,182 @@ func getServiceSpecByName(infraSpec *scaffold.InfraSpec, name string) *scaffold.
 }
 
-func printHintsAboutUseMongo(userResourceName string, usedResourceName string) {
-	log.Printf("'%s' uses '%s' will be achieved by these environment variables: ",
-		userResourceName, usedResourceName)
-	log.Printf("MONGODB_URL")
-	log.Printf("spring.data.mongodb.uri")
-	log.Printf("spring.data.mongodb.database")
-	log.Printf("Please make sure your application used the right environment variable.")
+func printHintsAboutUseMongo(userResourceName string, usedResourceName string,
+	console *input.Console, context *context.Context) {
+	(*console).Message(*context, fmt.Sprintf("CAUTION: '%s' uses '%s' will be achieved by these environment variables: ",
+		userResourceName, usedResourceName))
+	(*console).Message(*context, "MONGODB_URL")
+	(*console).Message(*context, "spring.data.mongodb.uri")
+	(*console).Message(*context, "spring.data.mongodb.database")
+	(*console).Message(*context, "Please make sure your application used the right environment variable name.\n")
 }
 
-func printHintsAboutUseCosmos(userResourceName string, usedResourceName string) {
-	log.Printf("'%s' uses '%s' will be achieved by these environment variables: ",
-		userResourceName, usedResourceName)
-	log.Printf("spring.cloud.azure.cosmos.endpoint")
-	log.Printf("spring.cloud.azure.cosmos.database")
-	log.Printf("Please make sure your application used the right environment variable.")
+func printHintsAboutUseCosmos(userResourceName string, usedResourceName string,
+	console *input.Console, context *context.Context) {
+	(*console).Message(*context, fmt.Sprintf("CAUTION: '%s' uses '%s' will be achieved by these environment variables: ",
+		userResourceName, usedResourceName))
+	(*console).Message(*context, "spring.cloud.azure.cosmos.endpoint")
+	(*console).Message(*context, "spring.cloud.azure.cosmos.database")
+	(*console).Message(*context, "Please make sure your application used the right environment variable name.\n")
 }
 
 func printHintsAboutUseServiceBus(userResourceName string, usedResourceName string,
-	isJms bool, authType internal.AuthType) error {
-	log.Printf("'%s' uses '%s' will be achieved by these environment variables: ",
-		userResourceName, usedResourceName)
+	isJms bool, authType internal.AuthType,
+	console *input.Console, context *context.Context) error {
+	(*console).Message(*context, fmt.Sprintf("CAUTION: '%s' uses '%s' will be achieved by these environment variables: ",
+		userResourceName, usedResourceName))
 	if !isJms {
-		log.Printf("spring.cloud.azure.servicebus.namespace")
+		(*console).Message(*context, "spring.cloud.azure.servicebus.namespace")
 	}
 	if authType == internal.AuthTypeUserAssignedManagedIdentity {
-		log.Printf("spring.cloud.azure.servicebus.connection-string=''")
-		log.Printf("spring.cloud.azure.servicebus.credential.managed-identity-enabled=true")
-		log.Printf("spring.cloud.azure.servicebus.credential.client-id")
+		(*console).Message(*context, "spring.cloud.azure.servicebus.connection-string=''")
```

Comment:
I am not comfortable with these specific properties are listing directly here, seems like scaffold_gen is a general place to generate the scaffold, but all these properties are only java and spring specific

Reply:
> but all these properties are only java and spring specific

No, properties like `POSTGRES_USERNAME` are used for all kinds of application, like .NET. The property list used in `scafold_gen.go` is make according to `resources.bicept`.

We can improve this logic in `scafole_gen.go` together with `resourcec.bicept` in this task:  [ADO#SJAD#2300813: User can use spring boot related features (like service connector and environment variables) only when the app is detected as spring boot app](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/2300813)